Source/JavaScriptCore/runtime/JSObject.cpp

```diff
@@ -1291,8 +1318,12 @@ ContiguousJSValues JSObject::convertUndecidedToContiguous(VM& vm)
     for (unsigned i = butterfly->vectorLength(); i--;)
         butterfly->contiguous().at(this, i).setWithoutWriteBarrier(JSValue());
 
+    Structure* oldStructure = structure();
     WTF::storeStoreFence();
```

Comment:
I take it that these two lines need to occur in order?

Reply:
No. Maybe, we should just move it after store-store-fence.

Reply:
The important thing is butterfly is fully updated when it is set. (this is the store-store-fence for that).